### PR TITLE
feat: optionally disable popping on backspace

### LINKED ISSUE #550: Feature request: Optionally disable popping of items on backspace keydown
**Is your feature request related to a problem? Please describe.**
Users are getting confused when trying to clear search filters with backspace and accidentally deselecting items. Using backspace on an empty search bar to deselect items doesn't seem to be expected behavior for most people. 

**Describe the solution you'd like**
Add a prop disablePoppingOnBackspace that defaults to false but is able to turn off this behavior.

**Describe alternatives you've considered**
Alternative could be disable by default and enable optionally but this would change existing behavior and likely wouldn't be ideal.

**Additional context**
```ts
//index.js

static propTypes = {
  //...
  tabIndex: PropTypes.number,
  disablePoppingOnBackspace: PropTypes.bool,
}

onKeyboardKeyDown = e => {
  //...
  } else if (!disablePoppingOnBackspace && e.key === 'Backspace' && tags.length && this.searchInput.value.length === 0) {
    const lastTag = tags.pop()
    this.onCheckboxChange(lastTag._id, false)
  } else {
    return
  }
  e.preventDefault()
}

//index.test.js
test('cannot delete tags on empty search input with backspace on keyboardNavigation with disablePoppingOnBackspace enabled', t => {
  const data = [{ ...node('a', 'a'), checked: true }, { ...node('b', 'b'), checked: true }]
  const wrapper = mount(<DropdownTreeSelect data={data} disablePoppingOnBackspace={true}/>)
  wrapper.instance().searchInput.value = 'x'
  triggerOnKeyboardKeyDown(wrapper, 'Backspace')
  t.deepEqual(wrapper.state().tags.length, 2)

  wrapper.instance().searchInput.value = ''
  triggerOnKeyboardKeyDown(wrapper, 'Backspace')
  t.deepEqual(wrapper.state().tags.length, 2)
  triggerOnKeyboardKeyDown(wrapper, 'Backspace')
  t.deepEqual(wrapper.state().tags.length, 2)
})
```

### PR DESCRIPTION
**Pull request by r-zane-spalding**
## What does it do?

Adds disablePoppingOnBackspace prop to allow for optionally disabling the popping of nodes from the tree when the 'backspace' key is pressed in an empty search bar.

## Fixes #550

Please mention in the format "Fixes #issueNumber" or "Closes #issueNumber".
This is important for semantic-release to correctly generate release tags and update issues.

## Type of change

Please delete options that are not relevant.

- [x ] New feature (non-breaking change which adds functionality)
- [ x] This change requires a documentation update

# Checklist:

- [ x] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] Updated documentation (if applicable)
- [ x] Added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes
- [ x] My changes generate no new warnings

### CODE REVIEW

File: src/index.keyboardNav.test.js [hunk 1]
```diff
@@ -146,6 +146,20 @@ test('can delete tags with backspace/delete on keyboardNavigation', t => {
...
+  const wrapper = mount(<DropdownTreeSelect data={data} disablePoppingOnBackspace={true} />)
+  wrapper.instance().searchInput.value = 'x'
+  triggerOnKeyboardKeyDown(wrapper, 'Backspace')
+  t.deepEqual(wrapper.state().tags.length, 2)
+
+  wrapper.instance().searchInput.value = ''
+  triggerOnKeyboardKeyDown(wrapper, 'Backspace')
+  t.deepEqual(wrapper.state().tags.length, 2)
+  triggerOnKeyboardKeyDown(wrapper, 'Backspace')
+  t.deepEqual(wrapper.state().tags.length, 2)
```
Comment (mrchief) [inline src/index.keyboardNav.test.js:160]: Curious, why the double test?

Comment (r-zane-spalding) [reply src/index.keyboardNav.test.js:160]: Good point, was following the pattern of the previous test but isn't really needed.

Comment (mrchief) [reply src/index.keyboardNav.test.js:160]: IOW, you don't have to create a new branch or PR everytime you make an edit

Comment (r-zane-spalding) [reply src/index.keyboardNav.test.js]: Gotcha, change made.

**Comment by mrchief** [comment]
@allcontributors[bot] Let's add @moonjy1993 for code

**Comment by mrchief** [comment]
@allcontributors Let's add @r-zane-spalding for code

**Review by mrchief — APPROVED** [review]

**Pull request merged**